### PR TITLE
NETOBSERV-1427 Added support to FLP dynamic configmap

### PR DESCRIPTION
**Pull request by OlivierCazade**
## Description

Added support to FLP dynamic configmap

## Dependencies

[FLP PR](https://github.com/netobserv/flowlogs-pipeline/pull/607)

## Checklist

If you are not familiar with our processes or don't know what to answer in the list below, let us know in a comment: the maintainers will take care of that.

* [X] Is this PR backed with a JIRA ticket? If so, make sure it is written as a title prefix _(in general, PRs affecting the NetObserv/Network Observability product should be backed with a JIRA ticket - especially if they bring user facing changes)._
* [ ] Does this PR require product documentation?
  * [ ] If so, make sure the JIRA epic is labelled with "documentation" and provides a description relevant for doc writers, such as use cases or scenarios. Any required step to activate or configure the feature should be documented there, such as new CRD knobs.
* [X] Does this PR require a product release notes entry?
  * [X] If so, fill in "Release Note Text" in the JIRA.
* [ ] Is there anything else the QE team should know before testing? E.g: configuration changes, environment setup, etc.
  * [ ] If so, make sure it is described in the JIRA ticket.
* QE requirements (check 1 from the list):
  * [X] Standard QE validation, with pre-merge tests unless stated otherwise.
  * [ ] Regression tests only (e.g. refactoring with no user-facing change).
  * [ ] No QE (e.g. trivial change with high reviewer's confidence, or per agreement with the QE team).

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -85,3 +85,5 @@ require (
 )
 
 replace github.com/prometheus/common v0.48.0 => github.com/netobserv/prometheus-common v0.48.0-netobserv
+
+replace github.com/netobserv/flowlogs-pipeline v0.1.12-0.20240322124726-d2b2352bfe0f => github.com/OlivierCazade/flowlogs-pipeline v0.0.2-0.20240325155329-387a8ce37ceb
```
Comment (OlivierCazade) [inline go.mod:89]: TODO: change this once the FLP PR is merged

File: controllers/flp/flp_common_objects.go [hunk 2]
```diff
@@ -282,7 +284,29 @@ func (b *builder) ConfigMap() (*corev1.ConfigMap, string, error) {
...
+		},
+		Data: map[string]string{
+			configFile: configStr,
+		},
+	}
+	hasher := fnv.New64a()
+	_, _ = hasher.Write([]byte(configStr))
+	digest := strconv.FormatUint(hasher.Sum64(), 36)
+	return &configMap, digest, nil
+}
```
Comment (jotak) [inline controllers/flp/flp_common_objects.go:307]: if I'm correct, the digest here isn't used, as digests are used to restart pods, and in the case of dynamic config we precisely don't want to restart pods?
If it's not used anywhere I'd rather not generate it, also to not bring confusion about how the reconciliation works

Comment (OlivierCazade) [reply controllers/flp/flp_common_objects.go]: Fixed, thanks.

**Comment by jotak** [comment]
thanks @OlivierCazade 
/lgtm

**Comment by memodi** [comment]
@OlivierCazade this PR has DNM label, is it ready for pre-merge testing?

**Comment by OlivierCazade** [comment]
@memodi Right now the PR FLP dependency is pointing to the associated PR dev branch. This PR is ready for testing but before merging we need to merge FLP PR first and then update the dependency to point to FLP main branch.

**Pull request merged**